src/modules/attributes/AttributesController.ts

```diff
@@ -349,28 +334,648 @@ export class AttributesController extends ConsumptionBaseController {
+
+        return [predecessor, successor]
+    }
+
+    /**
+     * Validators for attribute successions
+     */
+
+    public async validateRepositoryAttributeSuccession(
+        predecessorId: CoreId,
+        successorParams: IAttributeSuccessorParams
+    ): Promise<ValidationResult> {
+        const commonValidation = await this.validateAttributeSuccessionCommon(predecessorId, successorParams)
+        if (commonValidation.isError()) return commonValidation
+
+        const predecessor = (await this.getLocalAttribute(predecessorId))!
+        const successor = LocalAttribute.from({
+            id: CoreId.from(successorParams.id ?? "dummy"),
+            content: successorParams.content,
+            createdAt: successorParams.createdAt ?? CoreDate.utc(),
+            succeeds: successorParams.succeeds,
+            succeededBy: successorParams.succeededBy,
+            shareInfo: successorParams.shareInfo,
+            parentId: successorParams.parentId
+        })
+
+        if (!predecessor.isIdentityAttribute() || predecessor.isShared()) {
+            return ValidationResult.error(
+                CoreErrors.attributes.invalidPredecessor("Predecessor is not a valid repository attribute.")
+            )
+        }
+
+        if (!successor.isIdentityAttribute() || successor.isShared()) {
+            return ValidationResult.error(
+                CoreErrors.attributes.invalidPredecessor("Successor is not a valid repository attribute.")
+            )
+        }
+
+        return ValidationResult.success()
+    }
+
+    /**
+     * Validates the succession of an own shared identity attribute. Expects
+     * both predecessor and successor repository attributes to exist, as well as
+     * the predecessor's shared attribute copy.
+     */
+    public async validateOwnSharedIdentityAttributeSuccession(
+        predecessorId: CoreId,
+        successorParams: IAttributeSuccessorParams
+    ): Promise<ValidationResult> {
+        const commonValidation = await this.validateAttributeSuccessionCommon(predecessorId, successorParams)
+        if (commonValidation.isError()) return commonValidation
+
+        const predecessor = (await this.getLocalAttribute(predecessorId))!
+        const successor = LocalAttribute.from({
+            id: CoreId.from(successorParams.id ?? "dummy"),
+            content: successorParams.content,
+            createdAt: successorParams.createdAt ?? CoreDate.utc(),
+            succeeds: successorParams.succeeds,
+            succeededBy: successorParams.succeededBy,
+            shareInfo: successorParams.shareInfo,
+            parentId: successorParams.parentId
+        })
+
+        /* Predecessor must be an own shared identity attribute. */
+        if (
+            !predecessor.isIdentityAttribute() ||
+            !predecessor.isOwnSharedAttribute(this.identity.address) ||
+            predecessor.shareInfo.sourceAttribute === undefined
```

Comment:
these comments should be variables. e.g. ->

```suggestion
        const predecessorIsOwnSharedIdentityAttribute = !predecessor.isIdentityAttribute() ||
            !predecessor.isOwnSharedAttribute(this.identity.address) ||
            predecessor.shareInfo.sourceAttribute === undefined
        if (predecessorIsOwnSharedIdentityAttribute) {
```

Reply:
Pointless bikeshedding.

Reply:
Nope

Reply:
> Pointless bikeshedding.

Julian's right here. Comments are the root of all evil. They can get outdated quickly and therefore mislead the reader in the future. Imagine for example someone adds a new line in between the comment and the `if`. Now the comment describes something completely unrelated, and the reader doesn't have a clue what's going on. 

For reference: see book "Clean Code", subchapter "Don’t Use a Comment When You Can Use a Function or a Variable" (page 67). Actually, the whole chapter 4 is about comments and why they are mostly a bad idea.